### PR TITLE
K8s 1.31 provider slim support for s390x arch

### PR DESCRIPTION
**Pull request by chandramerla**
**What this PR does / why we need it**:
As with #1252 now k8s 1.30 provider slim supports s390x arch, now with this PR k8s provider **1.31 slim** support for s390x arch is added.
This PR needs to be merged once #1252 gets merged. Commit(s) https://github.com/kubevirt/kubevirtci/pull/1316/commits/493549619f6703f8ad2c294c30dc32c74becc5bd and later, would be mainly to be reviewed here, as all earlier commits in this PR are already reviewd and picked from #1252 (base for this PR)

**Special notes for your reviewer**:
@brianmcarey : The comments you've given on the PR #1252 for publish.sh are addressed in this PR.

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [x] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: publish.sh [hunk 1]
```diff
@@ -72,7 +69,7 @@ function build_clusters() {
 
       cluster-provision/gocli/build/cli provision --phases k8s cluster-provision/k8s/$i --slim
       ${CRI_BIN} tag ${TARGET_REPO}/k8s-$i ${TARGET_REPO}/k8s-$i:${KUBEVIRTCI_TAG}-slim
-    elif [ $ARCH == "s390x" ] && [ $i == "1.30" ]; then
+    elif [[ "$ARCH" == "s390x" && ( "$i" == "1.30" || "$i" == "1.31" ) ]]; then
```
Comment (brianmcarey) [inline publish.sh:72]: I guess these conditions can be removed when we include the k8s-1.32 provider and remove k8s-1.29 ?

**Review by brianmcarey — COMMENTED** [review]
/lgtm 

thanks @chandramerla - just one question inline

**Review by brianmcarey — COMMENTED** [review]
/approve

File: publish.sh (hunk 1 above)
Comment (chandramerla) [reply publish.sh:72]: right, I think whereever we use version of provider in publish.sh, I think we can remove them

**Pull request merged**